sdk/python/feast/feature_store.py

```diff
@@ -388,9 +392,17 @@ def get_online_features(
             {'sales:daily_transactions': [1.1,1.2], 'sales:customer_id': [0,1]}
         """
 
+        join_key_rows = []
+        for row in entity_rows:
+            join_key_row = {}
+            for key, value in row.items():
+                entity = self._registry.get_entity(key, self.project, allow_cache=True)
```

Comment:
Another nitpick, but since we are on `get_online_features()` i'd want us to be conscious of performance.

It seems a little bit unnatural that we are looping over `get_entity` twice, whether or not we are using the cached copy or not. The flow I was expecting was FeatureView -> Entity, use Entity to find Join Key. We have the FeatureViews as the requirements essentially, and this is what you implemented https://github.com/feast-dev/feast/pull/1429/files/c2d11871da3f9507a734d267f63b4719baf3df30..deb84094071761fd0eba3dc39ffd7ed7175b45fd#diff-ff4e2a65c8c29b7971edc77a61a7e970d410d77d9f173f72714d1bb49e2509a2R435

Ideally we'd not scan over all rows and for all keys in order to find the same entities. It seems more efficient to only call the registry once for each FeatureView+Entity combination, create a mapping of entity name to join_key, and reuse that mapping.

Reply:
good call - we now create a map from entity names to join keys that we reuse